### ISSUE TITLE
@raw doesn't work correctly with JavaScript 

### ISSUE THREAD

**Issue by lfmunoz** [casid/jte#180]
The example in the documentation shows:

```
@raw
<script>
  const foo = "foo";
  console.log("This is ${foo}");
</script>
@endraw
```

but are you sure that is valid JavaScript?  A template literal is using a ` not a "

When it generates the code it doesn't compile. In Kotlin the ${} is not escaped and that is a valid substitution in Kotlin so it says variable not found.  I.e,..

```
	jteOutput.writeContent("\n  const options = {\n      moduleCache: { vue: Vue },\n      async getFile(url) {\n        console.log(`getFifle - ${url}`)\n\n        const res = await fetch(url);\n        if ( !res.ok )\n          throw Object.assign(new
```

For Kotlin you probably should do triple quotes 	jteOutput.writeContent("""  """)

**Comment by casid** [comment]
Ugh, you are right, this is not valid javascript. I've adjusted the documentation.

There's similiar unit test for Kotlin (https://github.com/casid/jte/blob/4190ae2c0358229e1cb4e0ff617e16f07984da23/jte-kotlin/src/test/java/gg/jte/kotlin/TemplateEngineTest.java#L282), but it is nested within a content block. I'll try to add another test to reproduce your finding.

Thank you for reporting!

**Comment by casid** [comment]
@lfmunoz could it be that you do use an older jte version? I added a test case and it works fine. The escaping of $ for Kotlin was introduced in this commit: 1d7d63119687674ffacb6ab592f633dfe76d1b92

**Comment by lfmunoz** [comment]
Yes true I am using older version because my project is using Kotlin 1.5, the latest brings in 1.7 it seems and I get errors of incompatible versions. I tried to exclude 1.7 dependency in gradle but was having problems. Good to know it should work in latest thanks.

**Comment by casid** [comment]
The fix was first released with version `2.1.2`, that version did not yet include the Kotlin version bump. I think you should be able to upgrade without problems.

Version `2.2.1` caused the Kotlin incompatibility problems :-(

**Comment by casid** [comment]
@lfmunoz just a fyi, we did a downgrade of the used Kotlin version, so you should be able to safely upgrade to the latest jte version `2.2.2`.
